graphql_compiler/cost_estimation/int_value_conversion.py

```diff
@@ -102,7 +102,7 @@ def convert_field_value_to_int(
     if is_int_field_type(schema_info, vertex_class, property_field):
         return value
     elif is_datetime_field_type(schema_info, vertex_class, property_field):
-        return (value.astimezone(pytz.utc) - DATETIME_EPOCH_UTC) // datetime.timedelta(
+        return (value.replace(tzinfo=pytz.utc) - DATETIME_EPOCH_UTC) // datetime.timedelta(
```

Comment:
We used to treat naive times as local times. However, local times lose their meaning over RPC, so it's simpler to treat them as utc times.

Reply:
It seems that we are currently losing timezone information when converting datetime to ints. Can we not store the timezone information and use to it decode the value of datetime fields? Filtering with a timezone aware datetime on a timezone naive datetime field will likely lead to an error.

If you agree this is an issue feel free to address it in a separate PR or maybe just open an issue. I am also willing to help with it.

Reply:
That's not the issue. The stats are either in tz-aware or tz-naive format. Consider a query where the user gives argument in the opposite format (tz aware/naive). How do you compare these two datetimes?

If everything is happening on the same server (there's no RPC involved), then a tz-naive datetime means a datetime in local time. That can be converted to the tz of the tz-aware datetime, and they can be compared.

However, if you receive a request over RPC with tz-naive datetimes, you don't know the local timezone of the sender, and all you can do is assume it's utc.

Reply:
So yes, we are losing timezone information by assuming tz-naive datetimes are utc. However, it is an assumption we have to make, unless you have any suggestion.